### ISSUE TITLE
EnergyPlus v22.1 failed on writing output reports when the weather file folder contains some other files

### ISSUE THREAD

**Issue by jcyuan2020** [NREL/EnergyPlus#9601]
Issue overview
--------------
It was reported by a user that EnergyPlus will not finish reporting correctly for some unknown reason. This will happen to any idf file it runs (e.g. the simple example file 1ZoneUncontrolled3SurfaceZone.idf). The error message will contain a severe warning that seems to belong to the lower level std library: 
```
   ** Severe  ** invalid stoi argument
```

The html or csv report file will then just have a few lines of information: 
![image](https://user-images.githubusercontent.com/64040111/184658933-4a6dd74a-ef5d-4301-b73d-79a0a2157bb0.png)
![image](https://user-images.githubusercontent.com/64040111/184659200-dcf46ad0-881b-4643-95fd-f02c13aaac1f.png)

The command line running outputs seems to indicate that the calculation was finished, and at least the tabular output file writing stage is entered: 
```
Updating Shadowing Calculations, Start Date=12/07/2013
Continuing Simulation at 12/07/2013 for RUN PERIOD 1
Updating Shadowing Calculations, Start Date=12/27/2013
Continuing Simulation at 12/27/2013 for RUN PERIOD 1
Writing tabular output file results using comma format.
Writing tabular output file results using tab format.
Writing tabular output file results using text format.
Writing tabular output file results using HTML format.
Writing tabular output file results using XML format.
EnergyPlus Run Time=00hr 00min  1.11sec
Program terminated: EnergyPlus Terminated--Error(s) Detected.
```
The output variables (csv) report seems to be complete though. 

It was first suspected as an installation issue. However, later the user found that this is related to the weather file folder---if the folder where the weather (.epw) file is located contains some other types of files, the above scenario could occur. If the folder is cleaned to contain only the .epw file, then the problem will be gone and all the full complete reports can be obtained. 

The original workflow that produced the problem is that the user downloaded a weather file .zip from Onebuilding.org and extracted the compressed files into a new folder. The zip file contains a few other files (such as .clm, .ddy, .pvsyst, .rain, .stat, and .wea) files. The symptom can be reproduced when using v22.1 official release, using the weather file workflow and any example test file. 

So the work around for the v22.1 version is to clean up the weather file folder to contain only .epw file(s) and .ddy files. But a clean folder that contains only .epw type will also work. 

The good news is that with recent v22.2 pre-release develop branch, the problem is gone. It seems this problem has been unknowingly fixed by some PR that got merged after v22.1 official release. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): Windows (not tested on others)
 - Version of EnergyPlus (if using an intermediate build, include SHA): v22.1; not on recent v22.2 pre-release develop branch.
 - Helpdesk ticket number: 16229

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by Myoldmopar** [comment]
> The good news is that with recent v22.2 pre-release develop branch, the problem is gone. It seems this problem has been unknowingly fixed by some PR that got merged after v22.1 official release.

So...what's the point of this ticket?  We aren't going to re-release 22.1 with 22.2 a few weeks away.  If it's just to document the problem, then that's fine, just go ahead and close the ticket.  Then users of 22.1 can find this and get the workaround and status update.  If there's still work to be done in 22.2, then by all means leave it open.

**Comment by mjwitte** [comment]
@jcyuan2020 This looks like a duplicate of #9420, fixed in #9444

**Comment by jcyuan2020** [comment]
Thanks @mjwitte @Myoldmopar ! It's nice to see this problem has been purposely fixed and the root cause has been fully understood. And yes, anther reason for posting is to keep a record for the users of v22.1 since now it is still the latest official release version.
